File: WebViewApp/app/src/main/java/com/example/webviewapp/ui/fragment/NewsFragment.java
package com.example.webviewapp.ui.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONArray;
import com.alibaba.fastjson.JSONObject;
import com.example.webviewapp.R;
import com.example.webviewapp.common.adapters.NewsAdapter;
import com.example.webviewapp.data.NewsItem;
import com.example.webviewapp.databinding.FragmentNewsBinding;

import java.util.ArrayList;
import java.util.List;

public class NewsFragment extends Fragment {
    private static final String TAG = "NewsFragment";
    public FragmentNewsBinding viewBinding;
    String jsonString;

    private List<NewsItem> news;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        viewBinding = FragmentNewsBinding.inflate(inflater, container, false);
        viewBinding.newsList.setLayoutManager(new LinearLayoutManager(getActivity(), RecyclerView.VERTICAL, false));
        viewBinding.newsList.setAdapter(new NewsAdapter(news, getActivity(), R.layout.news_item));
        return viewBinding.getRoot();
    }

    public NewsFragment(String jsonString) {
        this.jsonString = jsonString;
        initNewsData(jsonString);
    }

    private void initNewsData(String jsonString) {
        news = new ArrayList<>();
        JSONObject json = JSON.parseObject(jsonString);
        JSONObject result = json.getJSONObject("result");
        JSONArray data = result.getJSONArray("data");
        for (int i = 0; i < data.size(); i++) {
            JSONObject object = data.getJSONObject(i);
            NewsItem item = new NewsItem();

            item.setUniquekey(object.getString("uniquekey"));
            item.setTitle(object.getString("title"));
            item.setCategory(object.getString("category"));
            item.setAuthorName(object.getString("author_name"));
            item.setUrl(object.getString("url"));
            item.setIsContent(object.getString("is_content"));

            for (String key : object.keySet()) {
                if (key.contains("thumbnail_pic_s")) {
                    item.getThumbnailPics().add(object.getString(key));
                }
            }

            news.add(item);
        }
    }
}

File: WebViewApp/app/src/main/java/com/example/webviewapp/ui/fragment/NewsFragment.kt
package com.example.webviewapp.ui.fragment

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.alibaba.fastjson.JSON
import com.alibaba.fastjson.JSONObject
import com.example.webviewapp.R
import com.example.webviewapp.common.adapters.NewsAdapter
import com.example.webviewapp.data.NewsItem
import com.example.webviewapp.databinding.FragmentNewsBinding
import java.util.*

class NewsFragment(private val jsonString: String) : Fragment() {
    var viewBinding: FragmentNewsBinding? = null
    private var news: MutableList<NewsItem>? = null
    fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        viewBinding = FragmentNewsBinding.inflate(inflater, container, false)
        viewBinding!!.newsList.setLayoutManager(LinearLayoutManager(getActivity(), RecyclerView.VERTICAL, false))
        viewBinding!!.newsList.setAdapter(NewsAdapter(news, getActivity(), R.layout.news_item))
        return viewBinding!!.root
    }

    private fun initNewsData(jsonString: String) {
        news = ArrayList()
        val json = JSON.parseObject(jsonString)
        Log.d(TAG, "initNewsData: $json")
        val result = json.getJSONObject("result")
        val data = result.getJSONArray("data")
        for (i in data.indices) {
            val `object` = data.getJSONObject(i)
            val item = NewsItem()
            initItem(`object`, item)
            for (key in `object`.keys) {
                if (key.contains("thumbnail_pic_s")) {
                    item.thumbnailPics.add(`object`.getString(key))
                }
            }
            news.add(item)
        }
    }

    private fun initItem(`object`: JSONObject, item: NewsItem) {
        item.uniquekey = `object`.getString("uniquekey")
        item.title = `object`.getString("title")
        item.category = `object`.getString("category")
        item.authorName = `object`.getString("author_name")
        item.url = `object`.getString("url")
        item.isContent = `object`.getString("is_content")
    }

    companion object {
        private const val TAG = "NewsFragment"
    }

    init {
        initNewsData(jsonString)
    }
}